### ISSUE TITLE
DurableOrchestrationClientBase appears to have internal type in public method return type

### ISSUE THREAD

**Issue by jakingmsft** [Azure/azure-functions-durable-extension#633]
Repo steps:

- Open Visual Studio 2017
- Install and reference Durable Task v1.7.1
- Try to implement `DurableOrchestrationClientBase`
- Create implementations for all `public` methods that need implemented
- Compiler will nag you that you need to also implement

`Task<OrchestrationStatusQueryResult> GetStatusAsync(DateTime createdTimeFrom, DateTime? createdTimeTo, IEnumerable<OrchestrationRuntimeStatus> runtimeStatus, int pageSize, string continuationToken, CancellationToken cancellationToken = default(CancellationToken))`

which it says is an `internal abstract` class.

It cannot be implemented because `OrchestrationStatusQueryResult` is `internal`. 

It is possible that this is a VS bug, as I wouldn't expect the `internal` to even be visible and it is illegal in C# to have an `internal` return type in a `public class`.

I'm curious how [this](https://github.com/Azure/azure-functions-durable-extension/blob/master/src/WebJobs.Extensions.DurableTask/DurableOrchestrationClientBase.cs) is able to build, but I haven't had time to check out the source code and try it.

Sincerely,
jaking

**Comment by cgillum** [comment]
Can you share your csproj file? I'm curious to see if non-matching nuget references could be the cause.

**Comment by jakingmsft** [comment]
@cgillum Here it is, the entire thing:

```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Autofac" Version="4.9.1" />
    <PackageReference Include="Microsoft.Azure.WebJobs.Extensions.DurableTask" Version="1.7.1" />
  </ItemGroup>

</Project>
```

Thank you for your help.

**Comment by ConnorMcMahon** [comment]
This is definitely a problem. I will be taking a look at this.

**Comment by ConnorMcMahon** [comment]
Addressing this in #635.

**Comment by cgillum** [comment]
Resolved with [v1.8.0](https://www.nuget.org/packages/Microsoft.Azure.WebJobs.Extensions.DurableTask/1.8.0) release.
